src/mame/heathkit/tlb.cpp

```diff
@@ -0,0 +1,688 @@
+	PORT_DIPSETTING(    0x06, "1800")
+	PORT_DIPSETTING(    0x07, "2000")
+	PORT_DIPSETTING(    0x08, "2400")
+	PORT_DIPSETTING(    0x09, "3600")
+	PORT_DIPSETTING(    0x0a, "4800")
+	PORT_DIPSETTING(    0x0b, "7200")
+	PORT_DIPSETTING(    0x0c, "9600")
+	PORT_DIPSETTING(    0x0d, "19200")
+	PORT_DIPNAME( 0x30, 0x00, "Parity")              PORT_DIPLOCATION("SW401:5,6")
+	PORT_DIPSETTING(    0x00, DEF_STR(None))
+	PORT_DIPSETTING(    0x10, "Odd")
+	PORT_DIPSETTING(    0x20, "None")
+	PORT_DIPSETTING(    0x30, "Even")
+	PORT_DIPNAME( 0x40, 0x00, "Parity Type")         PORT_DIPLOCATION("SW401:7")
+	PORT_DIPSETTING(    0x00, DEF_STR(Normal))
+	PORT_DIPSETTING(    0x40, "Stick")
+	PORT_DIPNAME( 0x80, 0x80, "Duplex")              PORT_DIPLOCATION("SW401:8")
+	PORT_DIPSETTING(    0x00, "Half")
+	PORT_DIPSETTING(    0x80, "Full")
+
+	PORT_START("SW402") // stored at 40C8
+	PORT_DIPNAME( 0x01, 0x00, "Cursor")             PORT_DIPLOCATION("SW402:1")
+	PORT_DIPSETTING(    0x00, "Underline")
+	PORT_DIPSETTING(    0x01, "Block")
+	PORT_DIPNAME( 0x02, 0x00, "Keyclick")           PORT_DIPLOCATION("SW402:2")
+	PORT_DIPSETTING(    0x02, DEF_STR(No))
+	PORT_DIPSETTING(    0x00, DEF_STR(Yes))
+	PORT_DIPNAME( 0x04, 0x00, "Wrap at EOL")        PORT_DIPLOCATION("SW402:3")
+	PORT_DIPSETTING(    0x00, DEF_STR(No))
+	PORT_DIPSETTING(    0x04, DEF_STR(Yes))
+	PORT_DIPNAME( 0x08, 0x00, "Auto LF on CR")      PORT_DIPLOCATION("SW402:4")
+	PORT_DIPSETTING(    0x00, DEF_STR(No))
+	PORT_DIPSETTING(    0x08, DEF_STR(Yes))
+	PORT_DIPNAME( 0x10, 0x00, "Auto CR on LF")      PORT_DIPLOCATION("SW402:5")
+	PORT_DIPSETTING(    0x00, DEF_STR(No))
+	PORT_DIPSETTING(    0x10, DEF_STR(Yes))
+	PORT_DIPNAME( 0x20, 0x00, "Mode")               PORT_DIPLOCATION("SW402:6")
+	PORT_DIPSETTING(    0x00, "VT52")
+	PORT_DIPSETTING(    0x20, "ANSI")
+	PORT_DIPNAME( 0x40, 0x00, "Keypad Shifted")     PORT_DIPLOCATION("SW402:7")
+	PORT_DIPSETTING(    0x00, DEF_STR(No))
+	PORT_DIPSETTING(    0x40, DEF_STR(Yes))
+	PORT_DIPNAME( 0x80, 0x00, "Refresh")            PORT_DIPLOCATION("SW402:8")
+	PORT_DIPSETTING(    0x00, "50Hz")
+	PORT_DIPSETTING(    0x80, "60Hz")
+INPUT_PORTS_END
+
+ROM_START( h19 )
+	ROM_REGION( 0x2000, "maincpu", ROMREGION_ERASEFF )
+#if USE_ALT_BIOS
+	ROM_DEFAULT_BIOS("H19")
+
+	ROM_SYSTEM_BIOS( 0, "H19", "H19" )
+	ROMX_LOAD( "2732_444-46_h19code.bin", 0x0000, 0x1000, CRC(f4447da0) SHA1(fb4093d5b763be21a9580a0defebed664b1f7a7b), ROM_BIOS(0) )
+
+	ROM_SYSTEM_BIOS( 1, "Super19", "Super19" )
+	ROMX_LOAD( "2732_super19_h447.bin", 0x0000, 0x1000, CRC(6c51aaa6) SHA1(5e368b39fe2f1af44a905dc474663198ab630117), ROM_BIOS(1) )
+
+	ROM_SYSTEM_BIOS( 2, "Watz19", "Watz19" )
+	ROMX_LOAD(  "watzman.bin", 0x0000, 0x1000, CRC(8168b6dc) SHA1(bfaebb9d766edbe545d24bc2b6630be4f3aa0ce9), ROM_BIOS(2) )
+
+	ROM_SYSTEM_BIOS( 3, "Ultra19", "Ultra19" )
+	ROMX_LOAD(  "2532_h19_ultra_firmware.bin", 0x0000, 0x1000, CRC(8ad4cdb4) SHA1(d6e1fc37a1f52abfce5e9adb1819e0030bed1df3), ROM_BIOS(3) )
```

Comment:
Won’t this fail validation due to the short names of the BIOS options not being all lowercase?